CMakeLists.txt

```diff
@@ -19,14 +22,16 @@ list(APPEND CMAKE_MODULE_PATH
 
 add_subdirectory(extern)
 
-option(BUILD_TESTING "Run unit tests only." OFF)
-
 if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
     include(CTest)
     add_definitions(-DUNITTEST)
     add_subdirectory(UNITTESTS)
 endif()
 
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_GREENTEA_TESTS)
+    enable_testing()
```

Comment:
Do we not want to `include(CTest)` if only `BUILD_GREENTEA_TESTS` is set (and not `BUILD_TESTING`)?

Reply:
We could `include(CTest)`, it defines some extra options that could be useful if we wanted to use CDash. 

However, we need to first set `BUILD_TESTING = ON` here. This is because we define the `BUILD_TESTING` option at the top of our root CMakeLists.txt and set it `OFF`. So when CTest.cmake defines `BUILD_TESTING` again, and gives it a default `ON` value, it takes no effect, as we have already created a cache entry with the same name. This effectively tells CMake to ignore CTest.cmake's `option` call (see https://cmake.org/cmake/help/latest/command/option.html).  Ultimately that means `enable_testing` isn't called when we `include(CTest)` because in CTest.cmake `enable_testing` is protected by `if(BUILD_TESTING)`.

I think the correct change here is to remove our definition of `BUILD_TESTING` and add the following to our top level CMakeLists.txt:
```
if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
    include(CTest)
endif()
```
Then if we're building mbed-os as the top level project, `BUILD_TESTING` will be `ON` by default (as set in CTest.cmake). This is how we set up greentea-client and is the standard way of doing things with CTest. I'll try making the change and see if I run into any other problems.